src/Microsoft.AspNetCore.Mvc.Core/Resources.resx

```diff
@@ -199,7 +199,7 @@
     <comment>{1} is the specific error message.</comment>
   </data>
   <data name="AttributeRoute_TokenReplacement_ReplacementValueNotFound" xml:space="preserve">
-    <value>While processing template '{0}', a replacement value for the token '{1}' could not be found. Available tokens: '{2}'.</value>
+    <value>While processing template '{0}', a replacement value for the token '{1}' could not be found. Available tokens: '{2}'. To use a '[' or a ']' as a literal string in a route or within a constraint use '[[' or ']]' instead.</value>
```

Comment:
The repeated "a" is a bit awkward and the sentence can be simplified. Suggest `To use a literal '[' or ']' in ...`. Also, add a comma after "constraint".